src/redis-check-aof.c

```diff
@@ -146,156 +266,412 @@ int readArgc(FILE *fp, long *target) {
     return readLong(fp,'*',target);
 }
 
-off_t process(FILE *fp) {
```

Comment:
btw, why did you have to refactor this function (and deal with the complexity of an output var for the `multi`?
isn't it easier to leave it as it was?

Reply:
I think the previous implementation is unreasonable. What I can remember is that in the original code, the `readAnnotations` function will use the while loop to read the annotations continuously until it finds that the following is not an annotation, and then uses `fseek` to go back to the previous position.

A more reasonable approach is that the outer caller is responsible for judging the type of the AOF line and calling the corresponding processing function according to the type. If the line starts with `*`, use `processRESP`, if it starts with `#`, call `readAnnotations`. Maybe in the future we will have other annotation types so it will be very easy for us to add a `readXXXAnnotations`.